tests/conftest.py

```diff
@@ -162,7 +149,7 @@ def pytest_collection_modifyitems(config, items):
 
 @pytest.fixture
 def fake_dist():
-    return FakeInstalledDistribution
+    return parse_fake_distribution
```

Comment:
```suggestion
    def _fake_dist(line, deps=[]):
        req = Requirement.parse(line)
        key = req.key
        if "==" in line:
            version = line.split("==")[1]
        else:
            version = "0+unknown"
        requires = [Requirement.parse(d) for d in deps]
        direct_url = None
        if req.url:
            direct_url = direct_url_from_link(Link(req.url))
        return Distribution(key, version, requires, direct_url)
    return _fake_dist
```

Thanks for refactor this. Wouldn't it makes sense to create a [fixture factory](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#factories-as-fixtures) since the `parse_fake_distribution` function only used by `fake_dist` fixture?